samples/KeyVault/deploy-eventhub-diagnostic-setting-for-keyvault/azurepolicy.json

```diff
@@ -0,0 +1,108 @@
+{
+   "properties": {
+       "mode": "all",
+       "displayName": "Deploy diagnostic setting for key vault to stream to event hub",
+       "description": "Automatically configure a diagnostic setting for key vault resources which will stream to a specified event hub namespace.",
+       "parameters": {
+           "diagnosticsSettingName": {
+               "type": "string",
+               "metadata": {
+                 "description": "Name for the diagnostic setting resource (e.g. 'SecOpsSendToEventHub')"
+               }
+           },
+           "eventHubAuthorizationRuleId": {
+               "type": "string",
+               "metadata": {
+                   "description": "Resource ID of the event hub authorization rule for the Event Hubs namespace in which the event hubs should be created and/or streamed to."
+               }
+           }
+       },
+       "policyRule": {
+           "if": {
+             "field": "type",
+             "equals": "Microsoft.KeyVault/vaults"
+           },
+           "then": {
+             "effect": "DeployIfNotExists",
+             "details": {
+               "type": "Microsoft.Insights/diagnosticSettings",
+               "name": "[parameters('diagnosticsSettingName')]",
```

Comment:
Checking existence by name technically works.  However, it could lead to you having multiple diagnosticSettings sending to the same event hub if a user already set it up manually.  How do you feel about removing the "name" condition and checking the `Microsoft.Insights/diagnosticSettings/eventHubAuthorizationRuleId` field in the "existenceCondition" instead?

I.e. 

```
"existenceCondition": {
   "field": "Microsoft.Insights/diagnosticSettings/eventHubAuthorizationRuleId",
   "equals": "[parameters('eventHubAuthorizationRuleId')]"
}
```

Reply:
That should work I think too, good suggestion.